packages/client/core/src/service/cryptid.ts

```diff
@@ -243,20 +261,25 @@ export class CryptidService {
       this.controllerChainPubkeys
     );
 
-    const middlewareResult = await this.executeMiddlewareInstructions(
-      account,
-      transactionAccountKeypair.publicKey,
-      "Propose"
-    );
+    // execute middleware if the transaction is proposed in "ready" state
+    // otherwise, wait until it is sealed
+    let middlewareResult: MiddlewareResult = { instructions: [], signers: [] };
+    if (state === TransactionState.Ready) {
+      middlewareResult = await this.executeMiddlewareInstructions(
+        account,
+        transactionAccountKeypair.publicKey,
+        "Propose"
+      );
+    }
 
     const unsignedProposeTransaction = await cryptidTransaction
-      .propose(this.program, transactionAccountKeypair.publicKey)
+      .propose(this.program, transactionAccountKeypair.publicKey, state)
       .signers(
```

Comment:
Good find. Strange that the double `.signers` call was not caught in a test. But I think we just don't have any middlewareResult with signers yet.